test/ServerSpec.js

```diff
@@ -96,50 +109,76 @@ describe('Unprotected routes: ', function() {
       });
     });
 
-    it('should return empty response for nonexistent users', function(done) {
-      request(loginOptions, function(err, res, body) {
+  });
+
+  describe('Memory creation', function() {
+   
+    // note: need aws credentials set up (in the current tab) for this to work
+    it ('should create a local file and push it to s3', function(done) {
+      // set timeout to 5 sec to give enough time to upload to s3
+      this.timeout(5000);
+      var form = new FormData();
+      form.append('type', 'image/png');
+      form.append('name', 'testImage.png');
+      form.append('memoryImage', fs.createReadStream('test/successkid.png'));
+
+      form.submit({
+        port: 3000,
+        path: '/api/memories/upload',
+        headers: {Authorization: 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI1ODBmYzdiMTZhYWE2ODM2OTk2NDc5MTQiLCJ1c2VybmFtZSI6Im5ldyIsInBhc3N3b3JkIjoidXNlciIsIl9fdiI6MCwibWVtb3JpZXMiOltdfQ.VfV0DtedVfOUZNAM6fOrMQCakF6Zrcbk-ujie0YGvd4'}
+        }, function(err, res) {
+          // res is the response to the form submission, not the response from the server (does not contain id)
+          if (err) {
+            console.log('err uploading image was ', err);
+          } 
+          expect(err).to.be.null;
+          expect(res.statusCode).to.equal(201);
+          done();
+      });
+    });
+  });
+
+  describe('Memory access', function() {
+    it ('should return the details of a memory given an id', function(done) {
+      // give it 5 sec to return all images
+      var accessOneOptions = {
+        method: 'GET',
+        url: 'http://localhost:3000/api/memories/id/5811455a578e4b766f7782f7',
```

Comment:
Should we artificially create a memory and use that returned ID to power this test? This test will fail if we delete all documents from the collection. A development database would preclude this problem